### PR TITLE
Feature/gmail data

### PR DESCRIPTION
**Pull request by willarion**
get gmail messages array with messages' data, added function to proceed GoogleMes to new convenient object added layout with list of messages

### CODE REVIEW

File: src/hooks/useMessages.ts [hunk 1]
```diff
@@ -1 +1,16 @@
-export {};
+import React from 'react';
+import { getMessagesList } from '../api/api';
+import { MessageToRender } from '../models/MessageToRender';
+
+function useMessages(token: string): Array<any> {
```
Comment (AnastasiaKozhemyachko) [inline src/hooks/useMessages.ts:5]: don't use any

File: src/hooks/useAuthToken.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { LoginContext } from '../pages/HomePage/HomePage';
+
+function useAuthToken(): string {
+  const { access } = React.useContext(LoginContext);
+  return access;
```
Comment (AnastasiaKozhemyachko) [inline src/hooks/useAuthToken.ts:6]: use can write `return React.useContext(LoginContext).access;`

File: src/utils/getUsefulMessageFields.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { GoogleMessage } from '../models/GoogleMessage';
+import { MessageToRender } from '../models/MessageToRender';
+
+interface Header {
+  name: string;
+  value: string;
+}
+
+export function getUsefulMessageFields(item: GoogleMessage): MessageToRender {
+  function filter(array: Header[], key: string) {
```
Comment (AnastasiaKozhemyachko) [inline src/utils/getUsefulMessageFields.ts:10]: more readable will be` Array<Header>`

**Review by AnastasiaKozhemyachko — CHANGES_REQUESTED** [review]

File: src/api/api.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { MessageToRender } from '../models/MessageToRender';
+import { getUsefulMessageFields } from '../utils/getUsefulMessageFields';
+
+const getMessages = async (
+  authToken: string,
+  // todo
+  // filter: unknown = {},
+  // pageToken?: string,
+): Promise<Array<string>> => {
+  const { data } = await axios.get(
```
Comment (bondarchukse) [inline src/api/api.ts:12]: You should use axios `param` property to send query params. see https://github.com/axios/axios

File: src/api/api.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const ids = await getMessages(authToken);
+  const messages = ids.map((id) => getMessageContent(id, authToken));
+  const result = await Promise.allSettled(messages);
+  return result
+    .filter((item) => item.status === 'fulfilled')
+    .map((item) => (item as PromiseFulfilledResult<GoogleMessage>).value)
+    .map(getUsefulMessageFields);
+};
+
+export { getMessagesList };
```
Comment (bondarchukse) [inline src/api/api.ts:47]: we are use `export const getMessagesList = ` syntax usually

File: src/components/Post/Post.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+interface PostsProps {
+  user: string;
+  title: string;
+  text: string;
+  dateAndTime: string;
+}
+
+function getProperDate(dateString: string): string {
```
Comment (bondarchukse) [inline src/components/Post/Post.tsx:10]: define it as arrow function. Try to keep the same coding style for all parts of application.

File: src/components/Post/Post.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  const date = new Date(dateString);
```
Comment (bondarchukse) [inline src/components/Post/Post.tsx:11]: you should check is `dateString` empty string, and is `date` valid

File: src/components/PostsContainer/PostsContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Post } from '../Post/Post';
+import useMessages from '../../hooks/useMessages';
+import useAuthToken from '../../hooks/useAuthToken';
+//todo import { NavigationButtons } from '../NavigationButtons/NavigationButtons';
+
+export const PostsContainer: React.FC = ({}) => {
+  const token = useAuthToken();
+  const messages = useMessages(token);
```
Comment (bondarchukse) [inline src/components/PostsContainer/PostsContainer.tsx:9]: you can simplify this part of the code with using `useAuthToken();`  inside the `useMessages` hook!

File: src/components/PostsContainer/PostsContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <section className="postsContainer" style={{ padding: '20px' }}>
```
Comment (bondarchukse) [inline src/components/PostsContainer/PostsContainer.tsx:11]: try to use scss instead of inline styles!

File: src/hooks/useMessages.ts [hunk 10]
```diff
@@ -1 +1,16 @@
...
+  const [messages, setMessages] = React.useState<Array<MessageToRender>>([]);
```
Comment (bondarchukse) [inline src/hooks/useMessages.ts:6]: `MessageToRender` name is little bit weird

File: src/hooks/useMessages.ts [hunk 11]
```diff
@@ -1 +1,16 @@
...
+  React.useEffect(() => {
+    getMessagesList(token)
```
Comment (bondarchukse) [inline src/hooks/useMessages.ts:8]: you will have this call on logout, for empty token!

File: src/utils/getUsefulMessageFields.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import { GoogleMessage } from '../models/GoogleMessage';
+import { MessageToRender } from '../models/MessageToRender';
+
+interface Header {
+  name: string;
+  value: string;
+}
+
+export function getUsefulMessageFields(item: GoogleMessage): MessageToRender {
```
Comment (bondarchukse) [inline src/utils/getUsefulMessageFields.ts:9]: try to refactor it with lodash functions
````
import { find } from 'lodash';

const titleHeader = find(item.payload.headers, { name: 'Subject' });
const title = titleHeader?.value;
````

**Pull request merged**